### ISSUE TITLE
Batched Newton-Schulz and Polar-Express normalization drops the reduced axes (raises, or silently wrong when batch == last dim)

### ISSUE THREAD

**Issue by mcgrof** [HomebrewML/HeavyBall#96]
## Summary

`zeropower_via_newtonschulz5` and `msign` both accept stacked inputs
(`assert G.ndim >= 2`, and the former is documented as the "batched Muon
implementation"), but their spectral normalization is

```python
G = G / G.norm(dim=(-2, -1)).clamp(min=eps)
```

`G.norm(dim=(-2, -1))` reduces both matrix axes **without** `keepdim`, so a
`[K, m, n]` input yields a shape `[K]` divisor. Broadcasting then aligns that
length-`K` vector against the **trailing** axis rather than the batch axis:

- `K == 1` still works, because a size-one divisor broadcasts against anything;
- `K > 1` and `K != n` raises;
- `K == n` succeeds and silently scales every **column** by a different
  matrix's norm.

The reference Muon implementation this is derived from keeps the axes
(`X / (X.norm(dim=(-2, -1), keepdim=True) + 1e-7)`), and the commented-out
line directly above `legacy_zeropower_via_newtonschulz5` in this same file
also carries `keepdim=True`.

## Reproducer

```python
import torch
import heavyball.utils as hbu

ns = hbu.zeropower_via_newtonschulz5.__wrapped__  # skip the compile wrapper

# 1) batch of one -> fine, which is why this can go unnoticed
g = torch.randn(1, 8, 6)
assert torch.allclose(ns(g.clone()), ns(g[0].clone())[None])

# 2) ordinary batch -> raises
try:
    ns(torch.randn(4, 8, 6))
except RuntimeError as e:
    print("raised:", e)
# raised: The size of tensor a (6) must match the size of tensor b (4)
#         at non-singleton dimension 2

# 3) batch size == trailing dim -> no error, wrong result
g = torch.randn(6, 8, 6)
batched = ns(g.clone())
looped = torch.stack([ns(g[i].clone()) for i in range(6)])
print("matches per-matrix:", torch.allclose(batched, looped, atol=1e-3))
# matches per-matrix: False   (max abs diff ~7e-2)
```

`msign` reproduces identically. Observed on heavyball 3.2.0 (PyPI) and on
`main` at e578d5c; PyTorch 2.9.1, though nothing here is version-specific.

## Why it matters in practice

Case 3 is the dangerous one: no error, no warning, just incorrectly
normalized orthogonalization results. Any code path that stacks same-shape
parameters into a leading batch dimension can land on a shape where the batch
count equals the trailing dimension, and a batch of one working correctly
makes the problem easy to miss in a smoke test.

## Fix

Add `keepdim=True` at both call sites (one word each).
`legacy_zeropower_via_newtonschulz5` asserts a 2-D input, where the norm is
already a scalar, so it is unaffected.

I'll post a PR for a fix next.
